content/en/docs/pipelines/multi-user.md

```diff
@@ -0,0 +1,188 @@
+
+<!---
+#### Configuring Google Cloud permissions for namespace
+#### Granting Google Cloud permissions to UI Artifact Preview and Visualization
+
+For artifacts stored on Google Cloud Storage (GCS), Kubeflow Pipelines deploys
+artifact fetcher servers on user namespaces. The servers use `default-editor`
+Kubernetes service account, so that they are using the same Google service
+account as pipeline runs.
+-->
+
+### When using the SDK
+
+* <!-- this should be in GCP pipelines - auth doc --> how to authenticate to the public endpoint
+
+When calling SDK methods for experiments, you need to provide the additional
+namespace argument. Runs, recurring runs are owned by an experiment. They are
+in the same namespace as the parent experiment, so you can just call their SDK
+methods same as before.
+
+For example:
+
+```python
+import kfp
+client = kfp.Client(...)
+
+client.create_experiment(name='<Your experiment name>', namespace='<Your namespace>')
+print(client.list_experiments(namespace='<Your namespace>'))
+client.run_pipeline(
+    experiment_id='<Your experiment ID>', # Experiment determines namespace.
+    job_name='<Your job ID>',
+    pipeline_id='<Your pipeline ID>')
+print(client.list_runs(experiment_id='<Your experiment ID>'))
+print(client.list_runs(namespace='<Your namespace>'))
+```
+
+We also provide a helper method that saves a user namespace as default context
+in config file `$HOME/.config/kfp/context.json`. After setting a default
+namespace, the SDK methods default to use this namespace if no `namespace`
+argument provided.
+
+```python
+import kfp
+client = kfp.Client(...)
+
+# Note, this saves the namespace in `$HOME/.config/kfp/context.json`. Therefore,
+# You only need to call this once. The saved namespace context will be picked up
+# by other clients you use later.
+client.set_user_namespace(namespace='<Your namespace>')
+print(client.get_user_namespace())
+
+client.create_experiment(name='<Your experiment name>')
+print(client.list_experiments())
+client.run_pipeline(
+    experiment_id='<Your experiment ID>', # Experiment determines namespace.
+    job_name='<Your job ID>',
+    pipeline_id='<Your pipeline ID>')
+print(client.list_runs())
+
+# Specifying a different namespace will override the default context.
+print(client.list_runs(namespace='<Your other namespace>'))
+```
+
+Detailed documentation for KFP SDK can be found on
+[Pipelines SDK Reference](https://kubeflow-pipelines.readthedocs.io/en/latest/source/kfp.client.html).
+
+### When using rest API or generated python API client
+
+Similarly, when calling rest API endpoints or using generated API client, namespace
+argument is required for experiment APIs. Note that namespace is referred to
+using a resource reference. The resource reference type should be `NAMESPACE`
+and resource reference key id should be namespace name.
+
+For example, we can use the generated python API client (kfp-server-api) like the following:
```

Comment:
Suggested revision:

The following example demonstrates how to use the generated Python API client (kf-server-api) in a multi-user environment.